dist/sunsynk-power-flow-card.js

```diff
@@ -162,6 +163,150 @@ class SunsynkPowerFlowCard extends LitElement {
+      capacity: this.hass.states[config.entities.prog6_capacity] || { state: '' },
+      charge: this.hass.states[config.entities.prog6_charge] || { state: '' }
+    };
+
+    let inverter_prog = {};
+    if (!config.entities.use_timer_248 || stateObj26.state == 'off') {
+      inverter_prog.show = 'no';
+    } else if (!config.entities.prog1_time 
+        || !config.entities.prog2_time 
+        || !config.entities.prog3_time 
+        || !config.entities.prog4_time
+        || !config.entities.prog5_time
+        || !config.entities.prog6_time
+      ) {
+      inverter_prog.show = 'no';
+    } else {
+      inverter_prog.show = 'yes';
+
+      const timer_now = new Date(); // Create a new Date object representing the current time
+      let prog_time1 = new Date(timer_now.getTime());
+      let prog_time2 = new Date(timer_now.getTime());
+      let prog_time3 = new Date(timer_now.getTime());
+      let prog_time4 = new Date(timer_now.getTime());
+      let prog_time5 = new Date(timer_now.getTime());
+      let prog_time6 = new Date(timer_now.getTime());
+
+      prog_time1.setHours(prog1.time.state.split(":")[0]);
+      prog_time2.setHours(prog2.time.state.split(":")[0]);
+      prog_time3.setHours(prog3.time.state.split(":")[0]);
+      prog_time4.setHours(prog4.time.state.split(":")[0]);
+      prog_time5.setHours(prog5.time.state.split(":")[0]);
+      prog_time6.setHours(prog6.time.state.split(":")[0]);
+      prog_time1.setMinutes(prog1.time.state.split(":")[1]);
+      prog_time2.setMinutes(prog2.time.state.split(":")[1]);
+      prog_time3.setMinutes(prog3.time.state.split(":")[1]);
+      prog_time4.setMinutes(prog4.time.state.split(":")[1]);
+      prog_time5.setMinutes(prog5.time.state.split(":")[1]);
+      prog_time6.setMinutes(prog6.time.state.split(":")[1]);
+
+      //Add one day
+      if (prog_time1 < timer_now) {
+        if (prog_time2 < timer_now) {
+          prog_time1.setTime(prog_time1.getTime() + (1*60*60*24*1000));
+        }
+      };
+      if (prog_time2 < timer_now) {
+        if (prog_time3 < timer_now) {
+          prog_time2.setTime(prog_time2.getTime() + (1*60*60*24*1000));
+        }
+      };
+      if (prog_time3 < timer_now) {
+        if (prog_time4 < timer_now) {
+          prog_time3.setTime(prog_time3.getTime() + (1*60*60*24*1000));
+        }
+      };
+      if (prog_time4 < timer_now) {
+        if (prog_time5 < timer_now) {
+          prog_time4.setTime(prog_time4.getTime() + (1*60*60*24*1000));
+        }
+      };
+      if (prog_time5 < timer_now) {
+        if (prog_time6 < timer_now) {
+          prog_time5.setTime(prog_time5.getTime() + (1*60*60*24*1000));
+        }
+      };
+      if (prog_time6 < timer_now) {
+        if (prog_time1 < timer_now) {
+          prog_time6.setTime(prog_time6.getTime() + (1*60*60*24*1000));
+        }
+      };
+
+      if (prog_time1 < timer_now && prog_time2 > timer_now) {
+        if ( prog1.charge.state === 'No Grid or Gen' ) {
```

Comment:
Are you using the kellarza integration ? We need to look at this sensor value is setup as a switch and sensor that returns a 0/1 if grid charge is enabled 

![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/7227275/d5376131-5c8d-4f22-bc43-70d75f3e7ec1)